sdk/Sdk.Analyzers/Extensions/TypeSymbolExtensions.cs

```diff
@@ -58,5 +61,83 @@ internal static string GetMinimalDisplayName(this ITypeSymbol type, SemanticMode
 
             return name;
         }
+
+        internal static bool IsIterableType(this ITypeSymbol typeSymbol, SymbolAnalysisContext context)
+        {
+            bool isArrayType = false;
+
+            if (typeSymbol is IArrayTypeSymbol)
+            {
+                if (string.Equals(typeSymbol.ToString(), typeof(byte[]).Name, StringComparison.OrdinalIgnoreCase))
+                {
+                    return false;
+                }
+
+                isArrayType = true;
+            }
+
+            bool IsIEnumerableTType = typeSymbol.IsOrImplementsOrDerivesFrom(context.Compilation.GetTypeByMetadataName(typeof(IEnumerable<>).FullName)!);
+
+            var IsIEnumerableType = typeSymbol.IsOrImplementsOrDerivesFrom(context.Compilation.GetTypeByMetadataName(typeof(IEnumerable).FullName)!);
+
+            return IsIEnumerableTType || IsIEnumerableType || isArrayType;
+        }
+
+        internal static bool IsOrImplementsOrDerivesFrom(this ITypeSymbol symbol, ITypeSymbol? other)
```

Comment:
Most of the methods in this class are taken from - [here](https://github.com/Azure/azure-functions-dotnet-worker/blob/main/sdk/Sdk.Generators/Extensions/ISymbolExtensions.cs).